lib/Driver/Driver.cpp

```diff
@@ -1650,6 +1650,31 @@ void Driver::buildOutputInfo(const ToolChain &TC, const DerivedArgList &Args,
 
   }
 
+  if (TC.getTriple().isOSWindows()) {
+    if (const Arg *A = Args.getLastArg(options::OPT_libc)) {
+      OI.RuntimeVariant =
+          llvm::StringSwitch<Optional<OutputInfo::MSVCRuntime>>(A->getValue())
+              .Cases("MD", "MultiThreadedDLL",
+                     OutputInfo::MSVCRuntime::MultiThreadedDLL)
+              .Cases("MDd", "MultiThreadedDebugDLL",
+                     OutputInfo::MSVCRuntime::MultiThreadedDebugDLL)
+              .Cases("MT", "MultiThreaded",
+                     OutputInfo::MSVCRuntime::MultiThreaded)
+              .Cases("MTd", "MultiThreadedDebug",
```

Comment:
I suggest including the actual library names as aliases, which will be more in line with a future `-libc=musl` or something. I'm not sure if the spelled-out names make sense or not, and if they *do* make sense I'd kind of suggest kebab-case instead of CamelCase (unless these are canonical names from MS). cc @troughton

Reply:
I think keeping `MD` etc. as one option is probably okay, but I think the full names aren't helpful descriptions, even if that's what Microsoft calls them. No-one is going to want a _non_-multithreaded version, and `MultiThreaded` is missing the most important piece of information: that it's static.

I've already stated my case for separating out static and debug into separate options, but if we're not doing that, something like `shared` or `shared-dll`, `shared-debug(-dll)`, and `static`/`static-debug` seems appropriate. If those names would collide with options for other platforms, I'd still favour a `windows` prefix. `-libc MultiThreaded doesn't tell me anything`; `-libc windows-static` tells me everything I need to know.

Reply:
The names that I have used are actually canonical names from Microsoft.  The long names are also canonical names from Microsoft.  It is the raw value used in Visual Studio and their project files.  The spellings are now reflected in CMake as well as of 3.15.  Furthermore, they are also the same spellings used by Intel in icc and ifort (yes, thats Intel's FORTRAN compiler on Windows).

I will have to take that `kebab-case` though, that is hilarious!

@troughton - I don't like the Windows in the name.  I'm happy having an alias of {shared,static}-{,debug-}ucrt though.

Reply:
{shared,static}-{,debug-}ucrt works for me.